x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtStringClaimValidator.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security.authc.jwt;
+
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.SignedJWT;
+
+import org.elasticsearch.ElasticsearchSecurityException;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.xpack.core.security.support.StringMatcher;
+
+import java.text.ParseException;
+import java.util.List;
+
+public class JwtStringClaimValidator implements JwtClaimValidator {
+
+    private final String claimName;
+    private final List<String> allowedClaimValues;
+    // Whether the claim should be a single string
+    private final boolean singleValuedClaim;
+    private final StringMatcher claimValueMatcher;
+
+    public JwtStringClaimValidator(String claimName, List<String> allowedClaimValues, boolean singleValuedClaim) {
+        this.claimName = claimName;
+        this.allowedClaimValues = allowedClaimValues;
+        this.singleValuedClaim = singleValuedClaim;
+        // if (allowedClaimValues.stream().anyMatch(v -> v.startsWith("/") || v.contains("*"))) {
+        // throw new ElasticsearchException("invalid allowed claim values, cannot use wildcard or regex");
+        // }
```

Comment:
What's going on here?

Reply:
This should have been removed. This code prevents wildcard or regex to be used for allow claim values. Today you can configure them using wildcard, but they are treated as literal string. I think it's trappy and not aligned to most other settings. The intention is to explicitly disallow them. But it's technical not "pure refactoring". So I meant to delete them and add back in a later PR. They are now removed.